### ISSUE TITLE
Trained Weights

### ISSUE THREAD

**Issue by reynoldscem** [liuzhuang13/DenseNetCaffe#3]
Hi. First off, thank you for so promptly porting this into Caffe, I assume this was directly because of the request on the user group?

I was wondering if you have trained this network (in caffe) yet, and whether there are weights available?

**Comment by liuzhuang13** [comment]
Yeah, several people asked for caffe version, so we wrote this but the code may be less reliable than Torch version since it's not thoroughly tested yet.

By weights I assume you want weights pretrained on ImageNet dataset. Right now we have a small model (#params=10M) in Torch format, the top 1 validation error is 28.7%. You can download it here  https://1drv.ms/u/s!AjwB4qLCejx-a17znBzqnquzaJY  This model takes 224 * 224 input

Converting a Torch model to a caffe one is possible, check this tool https://github.com/facebook/fb-caffe-exts#torch2caffe . We'll look into this tool later, since in the end we want to have pretrained models in different package available online, but if you want to use it urgently you may want to try it first.

**Comment by reynoldscem** [comment]
Thank you, that's great. I have some weights trained on CIFAR-10 (without augmentation) - if you would like me to share them, I can do so.

**Comment by liuzhuang13** [comment]
Yeah, I'm grateful that you actually trained densenets. But there was a bug when you used our code, see another issue: https://github.com/liuzhuang13/DenseNetCaffe/issues/3

How's the performance of your training on CIFAR? Given there was a bug I guess it might not be as good as our results in the paper.

Cheers

**Comment by reynoldscem** [comment]
Yep, I saw that issue, but unfortunately trained with the version before it was fixed. It has not affected the outcome to greatly. I get 12.45% error on the test set without any data aug for cifar-10.

**Comment by liuzhuang13** [comment]
Yeah, but compared to 7% reported in our paper that's not a very small gap. Of course, something else might affect the result too.

**Comment by reynoldscem** [comment]
Yeah, you're right. I will try training again from the updated code at some point, and report back.

**Comment by reynoldscem** [comment]
I must add that I used a batch size of 16 as I am training on a GTX 960. Our Titan cards are currently locked up training other networks.

This means that across the 100k iterations my network has seen 4 times fewer examples.

When I retrain I can try either running for a greater number of iterations, or average the gradients over 4 forward passes and see if it helps.

**Comment by liuzhuang13** [comment]
Ok, we'll see!
